invokeai/backend/stable_diffusion/diffusers_pipeline.py

```diff
@@ -619,6 +635,33 @@ def step(
         #     i.e. before or after passing it to InvokeAIDiffuserComponent
         latent_model_input = self.scheduler.scale_model_input(latents, timestep)
 
+        if (self.control_model is not None) and (kwargs.get("control_image") is not None):
```

Comment:
i haven't looked closely at how this works, but maybe this chunk of code can go inside `self.invokeai_diffuser.do_diffusion_step()`?